x-pack/legacy/plugins/apm/public/hooks/useLicense.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useEffect, useState } from 'react';
+import { ILicense } from '../../../../../plugins/licensing/public';
+import { useApmPluginContext } from './useApmPluginContext';
+import { License } from '../../../../../plugins/licensing/common/license';
+
+export function useLicense() {
+  // Set the initial licesnse to a new license with an "undefined license"
+  // signature. This allows us to always have a license defined.
+  const [license, setLicense] = useState<ILicense>(
+    new License({ signature: 'undefined license' })
+  );
+  const { license$ } = useApmPluginContext().plugins.licensing;
+
+  useEffect(() => {
+    const subscription = license$.subscribe(nextLicense => {
+      setLicense(nextLicense);
+    });
+    return () => subscription.unsubscribe();
+  }, [license$]);
```

Comment:
Shouldn't the subscription happen inside the context and not the other way around? If several components call `useLicense` we'll have several subscriptions. If we add it to the context we'll only have one.

Reply:
Fixed in b7e5f18bd0.